test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs

```diff
@@ -1,482 +1,494 @@
-
-//            foreach (string entry in _files.Keys.Union(_directories).Where(x => x.StartsWith(directoryName, StringComparison.Ordinal) || x.StartsWith(directoryName, StringComparison.Ordinal) && (x[directoryName.Length] == Path.DirectorySeparatorChar || x[directoryName.Length] == Path.AltDirectorySeparatorChar)))
-//            {
-//                string p = entry.Replace('\\', '/').TrimStart('/');
-//                if (g.IsMatch(p))
-//                {
-//                    yield return entry;
-//                }
-//            }
-//        }
-//    }
-
-//    public class MockGlobalRunSpec : IGlobalRunSpec
-//    {
-//        public IReadOnlyList<IPathMatcher> Exclude { get; set; }
-
-//        public IReadOnlyList<IPathMatcher> Include { get; set; }
-
-//        public IReadOnlyList<IPathMatcher> CopyOnly { get; set; }
-
-//        public IReadOnlyList<IOperationProvider> Operations { get; set; }
-
-//        public IVariableCollection RootVariableCollection { get; set; }
-
-//        public IReadOnlyList<KeyValuePair<IPathMatcher, IRunSpec>> Special { get; set; }
-
-//        public IReadOnlyDictionary<string, IReadOnlyList<IOperationProvider>> LocalizationOperations { get; set; }
-
-//        public string PlaceholderFilename { get; set; }
-
-//        public Dictionary<string, string> TargetRelativePaths { get; set; }
-
-//        public bool TryGetTargetRelPath(string sourceRelPath, out string targetRelPath)
-//        {
-//            return TargetRelativePaths.TryGetValue(sourceRelPath, out targetRelPath);
-//        }
-//    }
-
-//    public class MockMountPoint : IMountPoint
-//    {
-//        public MockMountPoint()
-//        {
-//            MockRoot = new MockDirectory("/", "/", this, null);
-//        }
-
-//        public MountPointInfo Info { get; set; }
-
-//        public IDirectory Root => MockRoot;
-
-//        public MockDirectory MockRoot { get; }
-
-//        public IFileSystemInfo FileSystemInfo(string fullPath)
-//        {
-//            string[] parts = fullPath.TrimStart('/').Split('/');
-
-//            IDirectory current = Root;
-
-//            for(int i = 0; i < parts.Length; ++i)
-//            {
-//                IFileSystemInfo info = current.EnumerateFileSystemInfos(parts[i], SearchOption.TopDirectoryOnly).FirstOrDefault();
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Abstractions.Mount;
+using Microsoft.TemplateEngine.Abstractions.PhysicalFileSystem;
+using Microsoft.TemplateEngine.Core.Contracts;
+using Microsoft.TemplateEngine.Core.Util;
+using Microsoft.TemplateEngine.TestHelper;
+using Microsoft.TemplateEngine.Utils;
+using Xunit;
+
+namespace Microsoft.TemplateEngine.Core.UnitTests
+{
+    public class MockFileStream : MemoryStream
+    {
+        private readonly Action<byte[]> _onFlush;
```

Comment:
consider Reactive's Subject in such future cases

- https://rehansaeed.com/reactive-extensions-part1-replacing-events/
- https://docs.microsoft.com/en-us/previous-versions/dotnet/reactive-extensions/hh229173(v=vs.103)